src/Node/Node.php

```diff
@@ -48,7 +51,13 @@ public function __construct(array $nodes = [], array $attributes = [], int $line
                 throw new \InvalidArgumentException(\sprintf('Using "%s" for the value of node "%s" of "%s" is not supported. You must pass a \Twig\Node\Node instance.', \is_object($node) ? \get_class($node) : (null === $node ? 'null' : \gettype($node)), $name, static::class));
             }
         }
-        $this->nodes = $nodes;
+        $this->nodes = [];
+        foreach ($nodes as $name => $node) {
+            if (!\is_int($name) && !is_string($name)) {
```

Comment:
array keys cannot be anything else that an int or a string, so this is dead code.

Reply:
Yes-ish. You can use a float even if you get a deprecation notice from PHP.

Reply:
you can use a float when *putting* a value in the array and it is deprecated. But you will not *read* a float as it is cast by the time you assign it

Reply:
Right